languages/cpp/generic/cpp_to_generic.ml

```diff
@@ -1230,9 +1239,11 @@ and map_exception_declaration tok env x : G.catch_exn =
   match x with
   | ExnDecl v1 -> (
       let v1 = map_parameter env v1 in
-      match H.parameter_to_catch_exn_opt v1 with
-      | Some x -> x
-      | None ->
+      match (H.parameter_to_catch_exn_opt v1, env.in_mode) with
+      | Some (CatchPattern (PatEllipsis t)), Target ->
+          CatchPattern (PatWildcard t)
```

Comment:
here actually when you have a CatchPattern PatEllipsis, Pattern, we want to generate a CatchPattern (PatEllispis).

This is maybe covered by your  Some x, _ case below, but I would put the explicit code and
a comment explaining it's important to transform differently catch(...) in Target vs Pattern mode.

Reply:
Done.